content/docs/user-guide/troubleshooting.md

```diff
@@ -125,3 +125,39 @@ and retry the DVC command. Specifically, one of:
 
 [internal directories]:
   https://dvc.org/doc/user-guide/project-structure/internal-files
+
+## DVC experiments may fail in shallow Git repositories {#git-shallow}
+
+DVC experiments rely on Git features which may not work properly in shallow Git
```

Comment:
Maybe include a link to https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt or elsewhere to explain shallow clones?